### ISSUE TITLE
Is there possibility to block third party cookies (block cross site tracking cookies and isolate other cross site cookies)

### ISSUE THREAD

**Issue by marcinkosztolowicz** [atlas-engineer/nyxt#3161]
**Is your feature request related to a problem? Please describe.**
I would like to start using Nyxt browser but I cannot find any information how to block third party cookies. 
The same feature as in Firefox block cross site tracking cookies and isolate other cross site cookies https://support.mozilla.org/en-US/kb/block-websites-storing-cookies-site-data-firefox 

I can see that there is possibility to create separate context , but in my opinion is not enough because it is hard to manage all context only to disable third party cookies tracking.

If there is possibility to turn on blocking third party cookies and have them isolated it would be beneficial to add this instruction to documentation https://nyxt.atlas.engineer/documentation

**Comment by aartaka** [comment]
Third party cookies are disabled by default. You can configure/force the behavior with
``` lisp
(define-configuration :browser
  "Forcing the no third party cookie behavior. 
Other options are: 
:NEVER to store no cookies at all.
:ALWAYS to accept all of them :/"
  ((default-cookie-policy :no-thirt-party)))
```

We also have enabled WebKit Intelligent Tracking Protection by default. Which is Apple's machine learning thingie trying to guess which websites need cookies and which should be blocked to hell. But this ITP thing is a black box, so no comments there.

Documentation is lacking, yes.

**Comment by jmercouris** [comment]
We could also post a Nyxt url to the page :-O

**Comment by marcinkosztolowicz** [comment]
Thank you for your answers! it was really informative. I hope maybe it will help someone in future, when she will search issues on github :) 

One thing is bothering me and I have some further questions.
I want to apologize if  I do not understand some concepts behind blocking third party cookies. 

What I observer
I can see cookies for third party site like youtube on different site where youtube video is embedded.

What I would like to have
At least I would like to have cookies with different user identification or even better to not have this third party cookies set.
(this is what firefox have)

How to reproduce

1. open Nyxt, go to youtube, open developer tools (function: open-inspector), checkout youtube cookies (eg, VISITOR_INFO1_LIVE)

![*_024](https://github.com/atlas-engineer/nyxt/assets/14169200/e4161380-d827-443d-9094-1d33d75abd54)

2.  open new tab and go to site with embedded youtube video (eg. [https://punchsalad.com/tutorials/embed-youtube-video/](https://punchsalad.com/tutorials/embed-youtube-video/) in this example you need to click play on video)
3. checkout cookies,  eg, VISITOR_INFO1_LIVE have the same value as in first point


![*_025](https://github.com/atlas-engineer/nyxt/assets/14169200/888c8876-ab81-41dc-9012-45421dfbb672)

**Comment by marcinkosztolowicz** [comment]
@aadcg  That was my first impression. I tested it with youtube as an third party and I thought that firefox handles it as I expected which is isolate or block those youtube cookies. I investigated it further and it looks like firefox does not block or isolate this youtube cookies so behavior is more or less similar to nyxt with :no-third-party configuration. 
I've also checked it with other cookies (probably from advertisers) and it looks like there are blocked as intended.
I am wondering why youtube cookies are not treated as third party cookies. 

To sum up. From nyxt perspective it is as good as firefox (with less configuration options in nyxt, but this is not the topic of this issue)
